File: examples/C/MultiGPU_openmp_example.c
#include <stdio.h>
#include <openacc.h>
#include <omp.h>
int main(int argc,char** argv)
{
    #pragma omp parallel 
    {
        int my_rank = omp_get_thread_num();
        // OpenACC Stuff
        #ifdef _OPENACC
        acc_device_t dev_type = acc_get_device_type();
        int num_gpus = acc_get_num_devices(dev_type);
        int my_gpu = my_rank%num_gpus;
        acc_set_device_num(my_gpu, dev_type);
        // We check what GPU is really in use
        my_gpu = acc_get_device_num(dev_type);
        // Alternatively you can set the GPU number with #pragma acc set device_num(my_gpu)
        printf("Here is thread %d: I am using GPU %d of type %d.\n", my_rank, my_gpu, dev_type);
        #endif  
    }
}

File: examples/Fortran/MultiGPU_openmp_example.f90
! you should add ` --option "-cpp" ` as argument to the idrrun command
program MultiGPU_openmp
    use ISO_FORTRAN_ENV, only : INT32
    use OMP_LIB
    use openacc
    implicit none    
    integer(kind=INT32)           :: my_rank
    integer                       :: num_gpus, my_gpu
    integer(kind=acc_device_kind) :: device_type

    !$omp parallel private(my_rank, my_gpu, device_type)
        my_rank = omp_get_thread_num()
        ! OpenACC Stuff
        #ifdef _OPENACC
        device_type = acc_get_device_type()
        num_gpus = acc_get_num_devices(device_type)
        my_gpu   = mod(my_rank,num_gpus)
        call acc_set_device_num(my_gpu, device_type)
        ! We check what GPU is really in use
        my_gpu = acc_get_device_num(device_type)
        ! Alternatively you can set the GPU number with #pragma acc set device_num(my_gpu)
        write(0,"(a14,i2,a17,i2,a9,i2)") "Here is thread ",my_rank," : I am using GPU ",my_gpu," of type ",device_type
        #endif  
    !$omp end parallel
end program MultiGPU_openmp

